src/sds.c

```diff
@@ -412,8 +412,13 @@ sds sdsResize(sds s, size_t size, int would_regrow) {
  * 4) The implicit null term.
  */
 size_t sdsAllocSize(sds s) {
-    size_t alloc = sdsalloc(s);
-    return sdsHdrSize(s[-1]) + alloc + 1;
+    char type = s[-1] & SDS_TYPE_MASK;
+    /* SDS_TYPE_5 header doesn't contain the size of the allocation */
+    if (type == SDS_TYPE_5) {
+        return s_malloc_size(sdsAllocPtr(s));
```

Comment:
I guess this is my only concern with this PR. IIUC before this PR this would have been a very slim and trivial operation to call sdsAllocSize for SDS_TYPE_5, but now this will become more expensive. I guess in your tests this is being compensated by the fact that we eliminated it in free, but in some cases (like evictions) we might experience overhead? 
Can we try and test CPU when heavy evictions are in place?

Reply:
Actually I see it is mainly used for AOF buf size. So I guess the concern is minor

Reply:
I guess my main point here is that we do not optimize SDS_TYPE_5 in this pr but we do impact the way we calculate it's allocation size. This makes sense from cleaner code maybe, but I wonder if changing it is something we want.

Reply:
anyway, since it DOES make the sdsResize and sdsFree code much cleaner I am in favor of approving this PR. (maybe @madolson will think differently?)

Reply:
Also note that TYPE5 will not be used for AOF. AOF buffer is initialized with an empty SDS to be expanded later. In this case we use TYP8 in the first place.

Reply:
I agree. I thought more about the future uses of sdsAllocSize

Reply:
It's maybe worth mentioning that no usage of sdsAllocSize() seems to require it to be precise EXCEPT for the new usages we're adding, which needs to be accurate for the free and determining space. We could add a "allocApproximate" and "allocExact". I don't feel strongly about it though, but wanted to float the idea.